src/expression_evaluator/expression_evaluator_visitor.cpp

```diff
@@ -57,6 +57,9 @@ void LambdaParamEvaluatorCollector::visit(ExpressionEvaluator* evaluator) {
         visit(child);
         visitSwitch(child);
     }
+    if (evaluator->getEvaluatorType() == EvaluatorType::LAMBDA_PARAM) {
```

Comment:
should we check this at the beginning of this function and return if the `if check` succeed?